### DISCUSSION TITLE
Global Script backup

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#6112] [Q&A]
Is there an easy way to programmatically backup the global scripts. We are starting to use that more often and, with several hands in there changing code, I'd like a way to keep a backup just in case someone mistakenly removes something that shouldn't be removed.

**Comment by pacmano1** [comment]
Like the Global Deploy Script?   I would leave a stub that calls a code template which could call other code templates.  Then properly manage your code templates.

**Comment by rutledj** [comment]
So, you are saying not to put any real code in the GS but rather in the code templates?
We are using it to store things like a list of valid account numbers for incoming lab results and then using that in the channel as a way to determine which lab and practice the amp is for.  

FYI, we have about 200 different practices across the country that we provide EHR services for. In some cases, we get labs from multiple lab companies through a common HIE channel were we have to use MSH values to direct them to the proper practice interface.  We are trying to get away from having to update the channels when we get new codes, so we started using the GS as a one stop place to do updates.

```
globalMap.remove('AtlasLabList');
	
	var atlasLabList = new Packages.java.util.ArrayList();
        var lab = new Object(); lab.ServiceArea='NANIN'; lab.LabOrg='ALVERNO'; lab.SendingApplication=''; lab.SendingFacility=''; lab.ReceivingApplication='UXN'; lab.ReceivingFacility='CEALVERNO'; lab.WorkQueue='NANINNWI'; atlasLabList.add(lab);
	var lab = new Object(); lab.ServiceArea='NANIN'; lab.LabOrg='ALVERNO'; lab.SendingApplication=''; lab.SendingFacility=''; lab.ReceivingApplication='UXO'; lab.ReceivingFacility='CEALVERNO'; lab.WorkQueue='NANINNWI'; atlasLabList.add(lab);

        var lab = new Object(); lab.ServiceArea='IKSIN'; lab.LabOrg='ALVERNO'; lab.SendingApplication=''; lab.SendingFacility=''; lab.ReceivingApplication='KYP'; lab.ReceivingFacility='CEALVERNO'; lab.WorkQueue=''; atlasLabList.add(lab);
	var lab = new Object(); lab.ServiceArea='IKSIN'; lab.LabOrg='ALVERNO'; lab.SendingApplication=''; lab.SendingFacility=''; lab.ReceivingApplication='KZL'; lab.ReceivingFacility='CEALVERNO'; lab.WorkQueue=''; atlasLabList.add(lab);

         ...

     globalMap.put('AtlasLabList', atlasLabList);

```

**Reply by pacmano1** [reply]
I would put those values in a database table and just source them to $g with a JS db script.

And you can just use a normal JSON blob (i.e. a JS object in $g) and call array prototype methods to get what you need. e.g.
```JS
var this_lab_receiving_facility = $g('AtlasLabList').find(ele => ele.ServiceArea == 'IKSIN').ReceivingFacility
```

**Comment by rutledj** [comment]
Thanks for the info. Unfortunately, we only have one person that is sql fluent. That would be a bottleneck for the other developers.

**Reply by pacmano1** [reply]
Well maybe think outside the box?    Create a channel that accepts the new row to insert and puts it to a database table and shields the team from the raw SQL  

And the insert (or upsert/merge) raw SQL or even a stored procedure you are talking about is trivial.  

 And developers with no SQL skills, really?    Anyway, I put my 2 cents in here, maybe others will chime in.

**Comment by rutledj** [comment]
I use the word "developers" loosely. They are actually Epic interface developers with enough mirth knowledge to create channels and do simple javascript code or basic mappings. Your suggestion of creating a channel to call a proc could be a possibility except that not all the global scripts follow the above example. Some are very simple.

**Reply by pacmano1** [reply]
I am off topic on arcitecture.   I just don't think what you are doing is a great way to optimize the engine or work.   Hopefully someone else will chime in.
